### PR TITLE
Create unique event counter for telemetries

### PR DESCRIPTION
**Pull request by NetanelBollag**
Unique event counter allow to count distinct events and send them as telemetry while using only anonymized and hashed data and send only the distinct count of the event. 

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR and in github.com/otterize/docs

### CODE REVIEW

File: src/shared/telemetries/telemetrysender/sender.go [hunk 1]
```diff
@@ -55,6 +70,58 @@ func (t *TelemetrySender) Send(component telemetriesgql.Component, eventType tel
...
+
+	for _, item := range batch {
+		t.uniqueEventsCounter.IncrementCounter(item.Event.Component, item.Event.EventType, item.Key)
+	}
+
+	counts := t.uniqueEventsCounter.Get()
+	logrus.Infof("Sending telemetry snapshot: %v", counts)
+	timeUntilReset := t.lastSnapshotResetTime.Add(t.snapshotResetInterval)
+
+	if time.Now().After(timeUntilReset) {
```
Comment (omris94) [inline src/shared/telemetries/telemetrysender/sender.go:109]: I think it should be at the end of the function 🤔

File: src/shared/telemetries/telemetrysender/config.go [hunk 2]
```diff
@@ -6,17 +6,19 @@ import (
...
+	TimeoutKey                            = "telemetry-client-timeout"
+	CloudClientTimeoutDefault             = "30s"
+	TelemetryEnabledKey                   = "telemetry-enabled"
+	TelemetryEnabledDefault               = false
+	TelemetryMaxBatchSizeKey              = "telemetry-max-batch-size"
+	TelemetryMaxBatchSizeDefault          = 100
+	TelemetryIntervalKey                  = "telemetry-interval-seconds"
+	TelemetryIntervalDefault              = 5
+	TelemetryAddressDefault               = "https://app.otterize.com/api/telemetry/query"
+	TelemetrySnapshotResetIntervalKey     = "telemetry-snapshot-reset-interval-duration"
```
Comment (omris94) [inline src/shared/telemetries/telemetrysender/config.go:19]: I think the name "Snapshot telemetry" might be a little bit intimidating.

File: src/shared/telemetries/telemetrysender/sender.go (hunk 1 above)
Comment (NetanelBollag) [reply src/shared/telemetries/telemetrysender/sender.go]: 100%, also I abort the reset if the send fail

File: src/shared/telemetries/telemetrysender/config.go (hunk 2 above)
Comment (NetanelBollag) [reply src/shared/telemetries/telemetrysender/config.go]: Fixed it to reset which is actually what it is

**Review by omris94 — APPROVED** [review]

**Pull request merged**